test/test_pysodium.py

```diff
@@ -177,12 +189,12 @@ def test_crypto_sign_sk_to_pk(self):
         self.assertEqual(pk, pk2)
 
     def test_AsymCrypto_With_Seeded_Keypair(self):
-        msg     = "correct horse battery staple"
```

Comment:
When I made the latest change, this test started failing in a nasty way. It turns we accept the string, pretend that it's bytes, and then get some garbage on the other end.

We should probably typecheck all of the functions that take a message, to make sure that the message is bytes. But I guess we'd also want to accept bytearray...what's the idiomatic way for checking that some object implements the buffer protocol?